### PR TITLE
daemon: repace goto exit with daemon_terminate()

### PR DESCRIPTION
**Pull request by ngortheone**
Start breaking down big main()
Remove goto exit label and replace it with a function that does cleanup.

### CODE REVIEW

**Comment by ngortheone** [comment]
cc @kevans91 @bsdimp

File: usr.sbin/daemon/daemon.c [hunk 1]
```diff
@@ -762,3 +755,19 @@ daemon_state_init(struct daemon_state *state)
 		.output_filename = NULL,
 	};
 }
+
+
+static _Noreturn void
+daemon_terminate(struct daemon_state *state)
+{
+	assert(state);
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:763]: Pointer should be explicitly compared against NULL when used like this

Comment (ngortheone) [reply usr.sbin/daemon/daemon.c:763]: ack, will fix in a bit

Comment (ngortheone) [reply usr.sbin/daemon/daemon.c]: fixed

**Comment by ngortheone** [comment]
committed by @kevans91  
https://reviews.freebsd.org/rGcf6356fd471cdf4e5c52550f63599881d722962c

**Pull request closed without merging**